src/main/java/de/tum/in/www1/artemis/web/rest/competency/CompetencyResource.java

```diff
@@ -555,20 +559,18 @@ public ResponseEntity<List<String>> getCourseCompetencyTitles(@PathVariable Long
     }
 
     /**
-     * Checks if the user has the necessary permissions and the competency matches the course.
+     * Checks if the competency matches the course.
      *
-     * @param role       The minimal role the user must have in the course
      * @param course     The course for which to check the authorization role for
      * @param competency The competency to be accessed by the user
      */
-    private void checkAuthorizationForCompetency(Role role, @NotNull Course course, @NotNull Competency competency) {
+    private void checkAuthorizationForCompetency(@NotNull Course course, @NotNull CourseCompetency competency) {
```

Comment:
I think you could update the name as well (this method no longer checks the authorization, but if the competency matches the course)